wasm: nil pointer deref when using unknown function

```
$ opa eval -t wasm 'sprinf("%s", [[ 1, 2, 3 ]])'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xbf5089]

goroutine 1 [running]:
github.com/open-policy-agent/opa/rego.(*Rego).compileWasm(0xc00036c280, 0x0, 0x0, 0x0, 0xc000391d00, 0x1, 0x1, 0x0, 0x0, 0x0, ...)
        /src/rego/rego.go:1332 +0x289
github.com/open-policy-agent/opa/rego.(*Rego).PrepareForEval(0xc00036c280, 0x1c21100, 0xc00003e1c0, 0x0, 0x0, 0x0, 0x1, 0x256fc78, 0x0, 0x0)
        /src/rego/rego.go:1457 +0x905
github.com/open-policy-agent/opa/cmd.eval(0xc00030f470, 0x1, 0x3, 0x0, 0xc000312570, 0x1, 0x1, 0x256fc78, 0x0, 0x0, ...)
        /src/cmd/eval.go:284 +0x9f5
github.com/open-policy-agent/opa/cmd.init.4.func2(0xc00031ef00, 0xc00030f470, 0x1, 0x3)
        /src/cmd/eval.go:220 +0xc5
github.com/spf13/cobra.(*Command).execute(0xc00031ef00, 0xc00030f440, 0x3, 0x3, 0xc00031ef00, 0xc00030f440)
        /src/vendor/github.com/spf13/cobra/command.go:766 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x24b9c20, 0xc000000180, 0xc00021ff78, 0x5d6945)
        /src/vendor/github.com/spf13/cobra/command.go:852 +0x2fe
github.com/spf13/cobra.(*Command).Execute(...)
        /src/vendor/github.com/spf13/cobra/command.go:800
main.main()
        /src/main.go:15 +0x31
```

The rego target output is
```
$ opa eval 'sprinf("%s", [[ 1, 2, 3 ]])'
{
  "errors": [
    {
      "message": "undefined function sprinf",
      "code": "rego_type_error",
      "location": {
        "file": "",
        "row": 1,
        "col": 1
      }
    }
  ]
}
```

--- a/rego/rego.go
+++ b/rego/rego.go
@@ -1440,6 +1440,10 @@ func (r *Rego) PrepareForEval(ctx context.Context, opts ...PrepareOption) (Prepa
 			},
 		},
 	})
+	if err != nil {
+		txnClose(ctx, err) // Ignore error
+		return PreparedEvalQuery{}, err
+	}
 
 	if r.target == targetWasm {
 

--- a/rego/rego.go
+++ b/rego/rego.go
@@ -1448,6 +1448,7 @@ func (r *Rego) PrepareForEval(ctx context.Context, opts ...PrepareOption) (Prepa
 	if r.target == targetWasm {
 
 		if r.hasWasmModule() {
+			txnClose(ctx, err) // Ignore error
 			return PreparedEvalQuery{}, fmt.Errorf("wasm target not supported")
 		}
 
@@ -1460,16 +1461,19 @@ func (r *Rego) PrepareForEval(ctx context.Context, opts ...PrepareOption) (Prepa
 
 		cr, err := r.compileWasm(modules, queries, evalQueryType)
 		if err != nil {
+			txnClose(ctx, err) // Ignore error
 			return PreparedEvalQuery{}, err
 		}
 
 		data, err := r.store.Read(ctx, r.txn, storage.Path{})
 		if err != nil {
+			txnClose(ctx, err) // Ignore error
 			return PreparedEvalQuery{}, err
 		}
 
 		o, err := opa.New().WithPolicyBytes(cr.Bytes).WithDataJSON(data).Init()
 		if err != nil {
+			txnClose(ctx, err) // Ignore error
 			return PreparedEvalQuery{}, err
 		}
 		r.opa = o

--- a/rego/rego_wasmtarget_test.go
+++ b/rego/rego_wasmtarget_test.go
@@ -57,6 +57,15 @@ func TestPrepareAndEvalWithWasmTarget(t *testing.T) {
 	}
 
 	assertPreparedEvalQueryEval(t, pq, []EvalOption{}, "[[true, true]]")
+
+	pq, err = New(
+		Query("foo(100)"),
+		Target("wasm"),
+	).PrepareForEval(ctx)
+
+	if err == nil {
+		t.Fatal("Expected error")
+	}
 }
 
 func TestPrepareAndEvalWithWasmTargetModulesOnCompiler(t *testing.T) {

